### ISSUE TITLE
[Bug] Grafana in container sets owner to "syslog:crontab" and fails to read config file

### ISSUE THREAD

**Issue by sandersaares** [grafana/grafana#9073]
Please include this information:
- What Grafana version are you using?
`grafana/grafana:latest`

- What datasource are you using?
N/A

- What OS are you running grafana on?
Ubuntu 16.04

- What did you do?
Prepared configuration and data directories for Grafana, setting root:root as the owner and root as the only read-write permissioned user, as is custom with Docker containers. Then started a Docker container with Grafana.

- What was the expected result?
Container starts successfully.

- What happened instead?
Container fails to start.

```
t=2017-08-22T07:35:03+0000 lvl=crit msg="Failed to parse /etc/grafana/grafana.ini, open /etc/grafana/grafana.ini: permission denied%!(EXTRA []interface {}=[])"
```

I can only guess that this may be because the owner of the files has been changed to syslog:crontab. This appears to happen on every Grafana container start.

On the host, I see this:

```
drw-rw---- 2 syslog crontab 4096 Aug 22 10:33 .
drwxr-xr-x 4 root   root    4096 Aug 22 10:33 ..
-rw-r--r-- 1 syslog crontab  788 Aug 22 10:31 grafana.ini
-rw-r--r-- 1 syslog crontab 2342 Aug 22 10:33 ldap.toml
```

http://docs.grafana.org/installation/docker/ has no instructions about filesystem permission configuration, so I do not know if I am in the wrong somehow or Grafana is.

The bottom line is that setting the volume owner to root:root with root as the only reader/wirter works for just about all other containers I have encountered, so I expect it to also work for Grafana. I certainly do not expect it to go changing file ownership!

**Comment by torkelo** [comment]
If you need to customize permissions you need to replace the run script so it runs as root as well:

https://github.com/grafana/grafana-docker/blob/master/run.sh#L42

**Comment by sandersaares** [comment]
It seems abnormal to me that Grafana requres its configuration file (containing passwords) to be readable by every user on the server (note that world-readable flag in the snippet above). Surely it should not require me to do some custom setup to make sure passwords can only be read by Grafana?

**Comment by torkelo** [comment]
It's not world readable for me, hmm maybe a docker thing or a side effect or your permission change?

**Comment by sandersaares** [comment]
Hmmm, yes I think I have to agree with your assessment. I looked on a different server with a different config and the world-readable flag is not getting set there. Must be something wrong with my setup.

**Comment by ewnd9** [comment]
Solution with custom Dockerfile resolved it for me https://github.com/grafana/grafana/issues/6285#issuecomment-254307611
